### ISSUE TITLE
[bug]: neutrino header verifier times out on mainnet proof verification/sync

### ISSUE THREAD

**Issue by jharveyb** [lightninglabs/taproot-assets#613]
### Background

Describe your issue here.

Mainnet tapd was offline for about a week. After start, I triggered a manual sync, which failed with error:

```sh
2023-10-23 16:13:24.392 [WRN] UNIV: encountered an error whilst syncing with server=(universe.ServerAddr) {
 ID: (int64) 1,
 addrStr: (string) (len=32) "universe.lightning.finance:10029",
 addr: (net.Addr) <nil>
}
: %!w(*fmt.wrapError=&{unable to register issuance proofs: unable to register issuance proofs: unable to verify issuance proofs: unable to verify proof: failed to validate proof block header: unable to retrieve block: rpc error: code = Unknown desc = did not get response before timeout 0xc0003d5f40})
```

Also observed a similar error from the scheduled background sync.

### Your environment

```sh
tapcli getinfo                                    # version of `tapd`, `lnd`, and network
uname -mrsv                                          # operating system 
bitcoind --version || btcd --version # version of `btcd`, `bitcoind`, or other backend
```
* any other relevant environment details

Testing PR https://github.com/lightninglabs/taproot-assets/pull/609 , lnd 0.17.0 with neutrino.

### Steps to reproduce

Leave tapd offline for a bit, then try to sync with the default universe.

**Comment by Roasbeef** [comment]
Looks like we make the header verifier here: https://github.com/lightninglabs/taproot-assets/blob/c7d57c597dfaadf0732a9ffb4bf63152c427f8f8/tapcfg/server.go#L129 

I think it's a timeout issue on the neutrino side of things. The block takes to long to fetch, or if all your peers are pruned then maybe they don't have the block.

**Comment by dstadulis** [comment]
https://github.com/lightningnetwork/lnd/pull/8111 will be used to retrieve the header information even if the neutrino backend doesn't have the full block.

**Comment by jharveyb** [comment]
Tracking changes here: #315.

**Comment by dstadulis** [comment]
fixed with #660
